test/conftest.py

```diff
@@ -0,0 +1,133 @@
+'''
+Utility functions used by the cibuildwheel tests.
+'''
+
+import subprocess, sys, os
+import pytest
+
+
+IS_WINDOWS_RUNNING_ON_AZURE = os.path.exists('C:\\hostedtoolcache')
+IS_WINDOWS_RUNNING_ON_TRAVIS = os.environ.get('TRAVIS_OS_NAME') == 'windows'
+
+platform = None
+
+if 'CIBW_PLATFORM' in os.environ:
+    platform = os.environ['CIBW_PLATFORM']
+elif sys.platform.startswith('linux'):
+    platform = 'linux'
+elif sys.platform.startswith('darwin'):
+    platform = 'macos'
+elif sys.platform in ['win32', 'cygwin']:
+    platform = 'windows'
+else:
+    raise Exception('Unsupported platform')
+
+
+@pytest.fixture
+def utils(tmp_path):
+    '''
+    Fixture use to run cibuildwheel with outputdir in a tmp_path, also list the wheels in that tmp_path
+    '''
+    class _utils:
+
+        def cibuildwheel_get_build_identifiers(self, project_path, env=None):
+            '''
+            Returns the list of build identifiers that cibuildwheel will try to build
+            for the current platform.
+            '''
+            cmd_output = subprocess.check_output(
+                [sys.executable, '-m', 'cibuildwheel', '--print-build-identifiers', project_path],
+                universal_newlines=True,
+                env=env,
+            )
+
+            return cmd_output.strip().split('\n')
+
+        def cibuildwheel_run(self, project_path, env=None, add_env=None):
+            '''
+            Runs cibuildwheel as a subprocess, building the project at project_path.
+
+            Uses the current Python interpreter.
+            Configure settings using env.
+            '''
+            if env is None:
```

Comment:
Is this `env` ever used?

Reply:
I think that It is needed in test which check that CIBW fail to start if no CI variable is set and `CIBW_PLATFORM` is not set. 

There is no such test now but I think that it should be created.

Reply:
It is done automatically by just running `cibuildwheel`, and test `01_basic` also tests it.

Reply:
No. Current version of `cibuildwheel` check on begin if it is running on some of CI or if `CIBW_PLATFORM` is set. But someone can suggest change which omit this check. 

Only possible option for check this in test is to unset some of environment variable. Otherwise after some change it may happen that launching tests will do some changes in host system.  Something like:
```
def test_platoform(utils):
    # check if cibuildwheel fail if runing without CIBW_PATFORM outside CI
    env = os.environ.copy()
    if "CIBW_PLATFORM" in env:
        del env["CIBW_PLATFORM"]
    if "CI" in env:
        del env["CI"]
    if "BITRISE_BUILD_NUMBER" in env:
        del env["BITRISE_BUILD_NUMBER"]
    if "AZURE_HTTP_USER_AGENT" in env:
        del env["AZURE_HTTP_USER_AGENT"]
    with pytest.raises(subprocess.CalledProcessError)
        utils.cibuildwheel_run()
```